boxsdk/auth/jwt_auth.py

```diff
@@ -360,3 +361,51 @@ def _normalize_rsa_private_key_passphrase(passphrase):
                 .format(passphrase.__class__.__name__)
             )
         return passphrase
+
+    @classmethod
+    def from_config_dictionary(cls, config_dictionary):
+        """
+        Create an auth instance as defined by the given configuration dictionary.
+
+        The dictionary should have the following structure:
+        {
+            "boxAppSettings": {
+                "clientID": "<client ID>",
+                "clientSecret": "<client secret>",
+                "appAuth": {
+                    "publicKeyID": "<public key ID>",
+                    "privateKey": "<RSA private key data>",
+                    "passphrase": "<private key passphrase>",
+                }
+            },
+            "enterpriseID": "<enterprise ID>",
+        }
+        :param config_dictionary:       Dictionary containing settings for configuring app auth.
+        :type config_dictionary:        `dict`
+        :return:                        Auth instance configured as specified by the config dictionary.
+        :rtype:                         :class:`JWTAuth`
+        """
+        if 'boxAppSettings' not in config_dictionary:
+            raise ValueError('boxAppSettings not present in configuration')
+        return cls(
+            client_id=config_dictionary['boxAppSettings']['clientID'],
+            client_secret=config_dictionary['boxAppSettings']['clientSecret'],
+            enterprise_id=config_dictionary.get('enterpriseID'),
+            jwt_key_id=config_dictionary['boxAppSettings']['appAuth']['publicKeyID'],
```

Comment:
I think all of the `appAuth` settings should use `['appAuth'].get('publicKeyID') or None` instead of `['appAuth']['publicKeyID']`. Three reasons for that:

1) If they forget to enter the correct values, the constructor will treat it as if the user didn't pass anything, instead of 0-length data (since the developer console defaults to showing these as empty strings).
2) In particular, it might be the case (though I'm not sure) that cryptography would treat an empty string passphrase differently from `None`.
3) This would allow the user to skip having to define `privateKey`, if they instead wanted to pass a system path (`rsa_private_key_file_sys_path`) via the keyword arguments I suggested.